### PR TITLE
Enhance the repo. 

### PR DESCRIPTION
**Pull request by CaiZhou2**
1. Change build system from setuptools to scikit-build-core.
2. Fix the docs-build-deploy error.
3. Enhance the repository's introductory sections, such as the README, LICENSE, tutorial cases, etc.
4. Add test cases. 
5. Code optimization.

### CODE REVIEW

File: .github/workflows/sphinx_docs_to_gh_pages.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     name: Deploy to GitHub Pages
     runs-on: ubuntu-latest    
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@master
```
Comment (ccaprani) [inline .github/workflows/sphinx_docs_to_gh_pages.yml:13]: Might be better to stick to a release version and not the ever changing master branch?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,44 +1,89 @@
...
+# Set macros
+option(BTLS "Build as a C++ terminal program" ON)
+option(PyBTLS "Build as a Python package" OFF)
+option(WIN_DEBUG "Debug BTLS on Windows" OFF)
+option(DEBUG_BUILD "Enable PyBTLS debug build" OFF)
+if(PyBTLS)
+  set(BTLS OFF)
 endif()
 
+# Set project name and version
```
Comment (ccaprani) [inline CMakeLists.txt:12]: Fantastic - have you tested build as a binary (non-python) locally? Does it work on just windows or more?

File: README.rst [hunk 3]
```diff
@@ -41,13 +41,15 @@ More detailed information about the input and output data can be found in the `M
...
-Credits
-=======
+Credits & Copyright
+===================
 PyBTLS is based on BTLS by Colin Caprani; 
 Modified by Ziyi Zhou; 
 Also contributed by Akbar Rizqiansyah. 
 
-Copyright 2024 The PyBTLS Developers.
+* Copyright (c) 2018-2021 Colin Caprani;
```
Comment (ccaprani) [inline README.rst:50]: I think 2001-2022.

File: README.rst [hunk 4]
```diff
@@ -41,13 +41,15 @@ More detailed information about the input and output data can be found in the `M
...
+* Copyright (c) 2022-2023 Colin Caprani, Ziyi Zhou, Akbar Rizqiansyah;
+* Copyright (c) 2024-? The PyBTLS developers (see contributors on GitHub).
```
Comment (ccaprani) [inline README.rst:52]: Just 2024 for now.

File: cpp/include/Vehicle.h [hunk 5]
```diff
@@ -15,6 +15,7 @@
 #include <memory>
 #include <math.h>
 #include "VehicleClassification.h"
+#include "BTLS_Config.h"
```
Comment (ccaprani) [inline cpp/include/Vehicle.h:18]: What's this and why required now?

File: cpp/src/py_main.cpp [hunk 6]
```diff
@@ -628,7 +638,7 @@ PYBIND11_MODULE(libbtls, m) {
 			.def("get_no_modes", &CMultiModalNormal::getNoModes);
 	py::class_<CDistribution> cdistribution(m, "_Distribution");
 		cdistribution.def(py::init<>())
-			.def(py::init<double, double, double>(), py::arg("w"), py::arg("m"), py::arg("s"))
+			.def(py::init<double, double, double>(), py::arg("loc"), py::arg("scale"), py::arg("shape"))
```
Comment (ccaprani) [inline cpp/src/py_main.cpp:641]: backwards compatible? Or doesn't affect the C++ code, just hte python argument names?

File: cpp/src/py_main.cpp [hunk 7]
```diff
@@ -646,6 +656,6 @@ PYBIND11_MODULE(libbtls, m) {
 			.def("gen_poisson", &CDistribution::GeneratePoisson)
 			.def("gen_gev", &CDistribution::GenerateGEV)
 			.def("gen_triangular", py::overload_cast<>(&CDistribution::GenerateTriangular))
-			.def("gen_triangular", py::overload_cast<double,double>(&CDistribution::GenerateTriangular), py::arg("mean"), py::arg("stdev"));
+			.def("gen_triangular", py::overload_cast<double,double>(&CDistribution::GenerateTriangular), py::arg("loc"), py::arg("w"));
```
Comment (ccaprani) [inline cpp/src/py_main.cpp:659]: ditto

File: docs/source/conf.py [hunk 8]
```diff
@@ -4,16 +4,6 @@
 # list see the documentation:
 # https://www.sphinx-doc.org/en/master/usage/configuration.html
```
Comment (ccaprani) [inline docs/source/conf.py:6]: suggest leave the path setup comments - it's from the boilerplate code

File: environment.yml [hunk 9]
```diff
@@ -3,7 +3,7 @@ channels:
   - conda-forge
   - defaults
 dependencies:
-  - python=3.11
+  - python=3.11.*
```
Comment (ccaprani) [inline environment.yml:6]: I don't think this should be necessary? 3.11 should be enough?

File: py/pybtls/__init__.py [hunk 10]
```diff
@@ -27,6 +18,7 @@
     Distribution,
     run,
     get_info,
+    __version__,  # synchronized with pyproject.toml
```
Comment (ccaprani) [inline py/pybtls/__init__.py:21]: Super!

File: py/pybtls/bridge/bridge.py [hunk 11]
```diff
@@ -97,10 +97,17 @@ def add_load_effect(
                     "The number of influence lines should be equal to the number of lanes."
                 )
 
+            if not all([inf_line._data_assigned for inf_line in inf_line_surf]):
```
Comment (ccaprani) [inline py/pybtls/bridge/bridge.py:100]: It would be better if these code changes were not included in a PR that redesigns the build system. Can you put these into a separate PR?

File: py/pybtls/garage/write.py [hunk 12]
```diff
@@ -53,6 +53,7 @@ def write_garage_file(
         else out_path.resolve().parent
     )
 
+    os.makedirs(absolute_out_dir, exist_ok=True)
```
Comment (ccaprani) [inline py/pybtls/garage/write.py:56]: Prefer Pathlib to os

File: py/pybtls/output/output_manager.py [hunk 13]
```diff
@@ -121,7 +121,7 @@ def read_data(
             "fatigue_events",
             "fatigue_rainflow",
         ],
-    ) -> list[pd.DataFrame]:
+    ) -> dict[str, pd.DataFrame]:
```
Comment (ccaprani) [inline py/pybtls/output/output_manager.py:124]: Again, code changes to a different PR?

File: pyproject.toml [hunk 14]
```diff
@@ -1,16 +1,16 @@
...
-build-backend = "setuptools.build_meta"
+build-backend = "scikit_build_core.build"
 
 [project]
 name = "pybtls"
-version = "0.3.7"  # Should also manually update in __init__.py
+version = "0.3.7"
 description = "A package for short-to-mid span bridge traffic loading simulation"
-readme = "README.md"
+readme = "README.rst"
```
Comment (ccaprani) [inline pyproject.toml:13]: I detest rst! Why not MD?

File: tests/test_bridge.py [hunk 15]
```diff
@@ -0,0 +1,70 @@
+import pytest
```
Comment (ccaprani) [inline tests/test_bridge.py:1]: Suggest tests could be a separate PR

**Review by ccaprani — CHANGES_REQUESTED** [review]
Fantastic work. Main comment is to split the changes up into different PRs:
- build system
- code changes
- enhancing test suite.

File: environment.yml (hunk 9 above)
Comment (CaiZhou2) [reply environment.yml:6]: Cannot create a new env from it on my Mac with only 3.11

File: docs/source/conf.py (hunk 8 above)
Comment (CaiZhou2) [reply docs/source/conf.py:6]: But it points to the local path without the *so file

File: py/pybtls/bridge/bridge.py (hunk 11 above)
Comment (CaiZhou2) [reply py/pybtls/bridge/bridge.py:100]: I will make changes based on your comments, then close this PR and create three new PRs.

File: pyproject.toml (hunk 14 above)
Comment (CaiZhou2) [reply pyproject.toml:13]: Hmmmm I just followed the pystra to make the pybtls readme...

File: cpp/src/py_main.cpp (hunk 6 above)
Comment (CaiZhou2) [reply cpp/src/py_main.cpp:641]: The previous argument names were typos. Now they match the unchanged C++ code.

File: cpp/src/py_main.cpp (hunk 7 above)
Comment (CaiZhou2) [reply cpp/src/py_main.cpp:659]: ditto

File: CMakeLists.txt (hunk 2 above)
Comment (CaiZhou2) [reply CMakeLists.txt:12]: Works as usual.

File: docs/source/conf.py (hunk 8 above)
Comment (ccaprani) [reply docs/source/conf.py:6]: just the boilerplate comment

**Comment by CaiZhou2** [comment]
Divided into 4 PRs

File: cpp/include/Vehicle.h (hunk 5 above)
Comment (CaiZhou2) [reply cpp/include/Vehicle.h]: Has been removed.
